ts/sagas/startup/checkSupportedFingerprintSaga.ts

```diff
@@ -0,0 +1,60 @@
+import { Effect } from "redux-saga";
+import { call, put, take } from "redux-saga/effects";
+import { getType } from "typesafe-actions";
+
+import { navigateToOnboardingFingerprintScreenAction } from "../../store/actions/navigation";
+import { fingerprintAcknowledge } from "../../store/actions/onboarding";
+
+import TouchID, { IsSupportedConfig } from "react-native-touch-id";
+
+export type BiometrySimpleType =
+  | "Fingerprint"
+  | "FaceID"
+  | "TouchID"
+  | "Not Enrolled"
+  | "Unavailable";
+
+export function* checkSupportedFingerprintSaga(): IterableIterator<Effect> {
+  // We check whether the user has already created a PIN by trying to retrieve
+  // it from the Keychain
+  const biometryTypeOrUnsupportedReason: BiometrySimpleType = yield call(
+    getFingerprintSettings
+  );
+
+  if (biometryTypeOrUnsupportedReason !== "Unavailable") {
+    yield put(
+      navigateToOnboardingFingerprintScreenAction({
+        biometryType: biometryTypeOrUnsupportedReason
+      })
+    );
+
+    // Here we wait the user accept the ToS
```

Comment:
wrong comment? it seems like we wait for fingerprintAcknowledge, not ToS

Reply:
fixed in e75e28b2